docs/conceptual/core/storage.md

```diff
@@ -2,8 +2,85 @@
 title: Storage
 ---
 
-* Key Value Database, RocksDB
+The Substrate uses a simple a key-value data store implemented as a
+database-backed modified Merkle tree.
 
-* Trie abstraction
+## Key-Value Database
 
-* Runtime Storage abstraction
+Substrate implements its storage database with [RocksDB](https://rocksdb.org/),
+a persistent key-value store for fast storage environments.
+
+This is used for all the components of Substrate which require persistent
+storage such as:
+
+- Substrate clients
+- Substrate light-clients
+- Off-chain workers
+
+## Trie Abstraction
+
+One advantage of using a simple key-value store is that you are able to easily
+abstract other storage structures on top.
+
+Substrate uses a Base-16 Modified Merkle Patricia tree ("trie") from
+[`paritytech/trie`](https://github.com/paritytech/trie) to provide a trie
+structure whose contents can be modified and whose root hash is recalculated
+efficiently.
+
+Tries are important tool for blockchains because they allow for efficient
```

Comment:
I think 'proof of state inclusion for data' is more important than state history property.
'Tries ar important tool for blockchains because they allow to produce and share small proof of data inclusion in a block. Those data are simple key values. Note that the trie structure also is a simple way to store historical of a block state.' .....

Reply:
In fact just adding the idea of 'provable' should be fine (my formulation is redundant with state trie part).

Reply:
I think I reviewed an old version, looks good here.